webapp/src/share.tsx

```diff
@@ -493,36 +497,46 @@ class SocialButton extends data.Component<SocialButtonProps, {}> {
     handleClick(e: React.MouseEvent<any>) {
         const { type, url: shareUrl, heading } = this.props;
 
-        let twitterText = lf("Check out what I made!");
         const socialOptions = pxt.appTarget.appTheme.socialOptions;
-        if (socialOptions.twitterHandle && socialOptions.orgTwitterHandle) {
-            twitterText = lf("Check out what I made with @{0} and @{1}!", socialOptions.twitterHandle, socialOptions.orgTwitterHandle);
-        } else if (socialOptions.twitterHandle) {
-            twitterText = lf("Check out what I made with @{0}!", socialOptions.twitterHandle);
-        } else if (socialOptions.orgTwitterHandle) {
-            twitterText = lf("Check out what I made with @{0}!", socialOptions.orgTwitterHandle);
-        }
-        const fbUrl = `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(shareUrl)}`;
-        const twitterUrl = `https://twitter.com/intent/tweet?url=${encodeURIComponent(shareUrl)}` +
-            `&text=${encodeURIComponent(twitterText)}` +
-            (socialOptions.hashtags ? `&hashtags=${encodeURIComponent(socialOptions.hashtags)}` : '') +
-            (socialOptions.related ? `&related=${encodeURIComponent(socialOptions.related)}` : '');
-
         pxt.tickEvent(`share.${type}`, undefined, { interactiveConsent: true })
 
         let url = '';
         switch (type) {
-            case "facebook": url = fbUrl; break;
-            case "twitter": url = twitterUrl; break;
+            case "facebook": {
+                url = `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(shareUrl)}`;
+                break;
+            }
+            case "twitter": {
+                let twitterText = lf("Check out what I made!");
+                if (socialOptions.twitterHandle && socialOptions.orgTwitterHandle) {
+                    twitterText = lf("Check out what I made with @{0} and @{1}!", socialOptions.twitterHandle, socialOptions.orgTwitterHandle);
+                } else if (socialOptions.twitterHandle) {
+                    twitterText = lf("Check out what I made with @{0}!", socialOptions.twitterHandle);
+                } else if (socialOptions.orgTwitterHandle) {
+                    twitterText = lf("Check out what I made with @{0}!", socialOptions.orgTwitterHandle);
+                }
+                url = `https://twitter.com/intent/tweet?url=${encodeURIComponent(shareUrl)}` +
+                    `&text=${encodeURIComponent(twitterText)}` +
+                    (socialOptions.hashtags ? `&hashtags=${encodeURIComponent(socialOptions.hashtags)}` : '') +
+                    (socialOptions.related ? `&related=${encodeURIComponent(socialOptions.related)}` : '');
+                break;
+            }
+            case "discourse": {
+                // https://meta.discourse.org/t/compose-a-new-pre-filled-topic-via-url/28074
```

Comment:
maybe better to use the script title as the "title=" and put the URL in "body=" ?

Reply:
and even if we don't have the title, I think the URL should be in the body

Reply:
I think `comments` would be a good icon

Reply:
Discourse automatically extracts the title from a URL pasted in the title.